recipes/json-schema-validator/all/conanfile.py

```diff
@@ -0,0 +1,74 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+from conans.tools import Version
+
+
+class JsonSchemaValidatorConan(ConanFile):
+    name = "json-schema-validator"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/pboettch/json-schema-validator"
+    description = "JSON schema validator for JSON for Modern C++ "
+    topics = ("json-schema-validator", "modern-json",
+              "schema-validation", "json")
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt"]
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    short_paths = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compiler_version = Version(self.settings.compiler.version)
+        if self.settings.os == "Windows"and self.settings.compiler == "Visual Studio" and compiler_version < "16":
+            tools.check_min_cppstd(self, "17")
+        elif self.settings.os == "Linux" and self.settings.compiler == "clang" and compiler_version < "4":
+            tools.check_min_cppstd(self, "11")
+        elif self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "11")
```

Comment:
```suggestion
        if self.settings.get_safe("compiler.cppstd"):
            tools.check_min_cppstd(self, "17")
        compilers = {"gcc": "7", "clang": "5", "Visual Studio": "15", "apple-clang": "10"}
        min_version = compilers.get(str(self.settings.compiler))
        if not min_version:
        	self.output.warn("{} recipe lacks information about the {} compiler support.".format(self.name, self.settings.compiler))
        else:
            if tools.Version(self.settings.compiler.version) < min_version:
                raise ConanInvalidConfiguration("{} requires c++17 support. The current compiler {} {} does not support it.".format(self.name, self.settings.compiler, self.settings.compiler.version))
```

Reply:
Sadly C++17 is only required for Windows, the rest accepted C++11.

Perhaps I should apply a patch? https://github.com/pboettch/json-schema-validator/pull/107

Reply:
that patch is not bad idea, but wait for the upstream first. Otherwise, just use some conditional to check the compiler.

Reply:
applied suggestion